core/src/test/scala/org/apache/spark/sql/test/generator/IndexColumn.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.test.generator
+
+import org.apache.spark.sql.test.generator.DataType.{getBaseType, ReflectedDataType}
+import org.apache.spark.sql.test.generator.TestDataGenerator.{getDecimal, getLength}
+
+trait IndexColumn {
+  val id: Int
+  def getId: Int = id - 1
+  def getLength: Integer = null
+}
+
+case class DefaultColumn(id: Int) extends IndexColumn {}
+
+case class PrefixColumn(id: Int, length: Int) extends IndexColumn {
+  override def getLength: Integer = length
+}
+
+case class ColumnInfo(columnName: String,
+                      dataType: ReflectedDataType,
+                      length: (Integer, Integer),
+                      desc: String) {
```

Comment:
String may be problematic. How about an enumeration?

Reply:
that is a description string, as it is named.

Reply:
what if user passes desc like "adededed"?

Reply:
invalid input (may be ignored). Since the description string is only used by generator, it is easy to assure its correctness

Reply:
okay. String seems ok too.